extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -76,6 +76,22 @@ export class ApplicationInsights extends BaseTelemetryPlugin implements IAppInsi
         config.enableAutoRouteTracking = stringToBoolOrDefault(config.enableAutoRouteTracking);
         config.namePrefix = config.namePrefix || "";
 
+        config.maxBatchSizeInBytes = config.maxBatchSizeInBytes > 0 ? config.maxBatchSizeInBytes : 102400; // 100kb
```

Comment:
Why are we doing all of these?

Setting the defaults for lines 80, 82 I get as they are not provided otherwise, but the other checks to "set" the existing value strictly to a boolean or like on line 88 to the same value doesn't seem right...

It is also really bad from a minification perspective as all of the long names on the config elements can't be compressed.

Reply:
ok, just went and looked at the bug and the implementation...
As the usage within the code iterates over the list of "keys" on the object returned by getDefaultConfig(), this sort of works -- it's just not very nice.

Reply:
Ok, so this line ```_self.config = ctx.getExtCfg<IConfig>(identifier)``` (Line 498 on my repo) deals with the "identifier" version of things as this will just return those values.... So it's just the root level ones that we are interested in that are getting missed (which is what the default stuff is effectively doing)...

I'm wondering whether we can do something really simple like... (_assignFn comes form the shims)
_self.config = __assignFn(ApplicationInsights.getDefaultConfig(), config, ctx.getExtCfg<IConfig>(identifier));